### ISSUE TITLE
Gtk exception when hiding deleted episodes

### ISSUE THREAD

**Issue by auouymous** [gpodder/gpodder#727]
Click show all episodes in view menu. Select a channel that only has deleted episodes. Select one of its episosdes. You might also need to lock it and then unlock it to force an update. Click hide deleted episodes in view menu. Gtk throws a ValueError exception in gtkui/model.py update().

```
model.set(model.get_iter((self.index,)), *(x for fields in (base_fields, update_fields)
                                           for pair in fields for x in pair))
```

And selecting another channel throws a Gtk warning.

```
/usr/lib/python3.6/site-packages/gpodder/gtkui/model.py:322: Warning: Source ID 960 was not found when attempting to remove it
  GObject.source_remove(self.background_update_tag)
```

I think it happens because the TreeView widget is removed and update() attempts to access it. Is it possible to determine if the channel will be hidden and select the "All episodes" view?

**Comment by elelay** [comment]
Indeed, this might be an untested case... I'll look into it

**Comment by elelay** [comment]
I can't reproduce it.

**Comment by auouymous** [comment]
I forgot to mention that the channel must be in a category that will be hidden when you hide deleted episodes.

**Comment by elelay** [comment]
Tried on an empty gpodder home:
 - added a single youtube channel, downloaded 1 video, deleted it
 - marked all episodes old
 - checked View > hide podcasts without episodes
 - selected the downloaded then deleted episode
 - changed the view to unplayed episodes
 - 'Video' category is hidden, nothing in the podcast and episode trees but no error.

**Comment by auouymous** [comment]
What about with two categories, the top most will be hidden but not any categories below it. It normally selects the next non-hidden channel in the current category. But when the entire category is hidden, it throws error instead of selecting into the next category. It might not throw the error if the categories below current category are also hidden, I'll test that tomorrow.

**Comment by auouymous** [comment]
Okay, it is a weird error. It happens when the last channel in a category (A) is hidden AND the category (B) below it is also hidden AND any category (C) below (B) will not be hidden. Category (A) does not need to hide for the error to occur. It doesn't matter if categories above (A) have downloaded or deleted episodes.

For categories A, B and C. Download an episode for a channel in C. Channels in B must only have deleted episodes, and last channel in A must only have deleted episodes. Lock and unlock a deleted episode for last channel in A. Then hide deleted episodes.

**Comment by elelay** [comment]
I'm sorry but I repeated your steps and couldn't trigger the exception.
Anyway I believe you :-)
A fix is welcome if you can manage it...

**Comment by elelay** [comment]
Now I can reproduce it by typing in podcast tree to filter, for my particular set of podcasts right now.
Would you accept this fix (catch the ValueError and don't update the model when it happens)?
```
            try:
                it = model.get_iter((self.index,))
            # fix #727 the tree might be invalid when trying to update so discard the exception
            except ValueError:
                break
            model.set(it, *(x for fields in (base_fields, update_fields)
                              for pair in fields for x in pair))
```
It seems to work fine for me in this case.

**Comment by auouymous** [comment]
That seems to have fixed it, thanks!
